modules/db/src/test/java/org/mule/module/db/internal/processor/PreparedBulkUpdateMessageProcessorDebugInfoTestCase.java

```diff
@@ -0,0 +1,119 @@
+    public static final QueryTemplate QUERY_TEMPLATE_WITH_NAMED_PARAM = new QueryTemplate(QUERY_SQL, UPDATE, Collections.<QueryParam>singletonList(new DefaultInputQueryParam(1, null, null)));
+    public static final String PARAM1 = PARAM_DEBUG_FIELD_PREFIX + 1;
+    public static final String PARAM_SET1 = PARAM_SET_DEBUG_FIELD_PREFIX + 1;
+    public static final String PARAM_SET2 = PARAM_SET_DEBUG_FIELD_PREFIX + 2;
+
+    private final MuleEvent event = mock(MuleEvent.class);
+    private final MuleMessage message = mock(MuleMessage.class);
+    private final DbConnection connection = mock(DbConnection.class);
+    private final DbConnectionFactory dbConnectionFactory = mock(DbConnectionFactory.class);
+    private final DbConfigResolver dbConfigResolver = mock(DbConfigResolver.class);
+    private final DbConfig dbConfig = mock(DbConfig.class);
+    private final QueryResolver queryResolver = mock(QueryResolver.class);
+    private final MuleContext muleContext = mock(MuleContext.class, RETURNS_DEEP_STUBS);
+
+    @Test
+    public void returnDebugInfo() throws Exception
+    {
+        List<String> planetNames = new ArrayList<>();
+        planetNames.add(EARTH.getName());
+        planetNames.add(MARS.getName());
+
+        when(message.getPayload()).thenReturn(planetNames);
+        when(event.getMessage()).thenReturn(message);
+
+        when(dbConnectionFactory.createConnection(TransactionalAction.NOT_SUPPORTED)).thenReturn(connection);
+
+        when(dbConfigResolver.resolve(event)).thenReturn(dbConfig);
+        when(dbConfig.getConnectionFactory()).thenReturn(dbConnectionFactory);
+
+        when(queryResolver.resolve(argThat(equalTo(connection)), any(MuleEvent.class))).thenReturn(new Query(QUERY_TEMPLATE_WITH_NAMED_PARAM));
+
+        final ParamValueResolver paramValueResolver = mock(ParamValueResolver.class);
+        when(paramValueResolver.resolveParams(any(MuleEvent.class), any(List.class))).thenReturn(Collections.singletonList(new QueryParamValue(null, EARTH.getName()))).thenReturn(Collections.singletonList(new QueryParamValue(null, MARS.getName())));
+
+        PreparedBulkUpdateMessageProcessor processor = new PreparedBulkUpdateMessageProcessor(dbConfigResolver, queryResolver, null, NOT_SUPPORTED, Collections.singletonList(UPDATE), paramValueResolver);
+        processor.setMuleContext(muleContext);
+
+        final List<FieldDebugInfo> debugInfo = processor.getDebugInfo(event);
+
+        assertThat(debugInfo.size(), equalTo(2));
+
+
+        List<FieldDebugInfo> queryFields = new ArrayList<>();
+        queryFields.add(createFieldDebugInfo(SQL_TEXT_DEBUG_FIELD, String.class, QUERY_SQL));
+        queryFields.add(createFieldDebugInfo(TYPE_DEBUG_FIELD, String.class, UPDATE.toString()));
+        assertThat(debugInfo, hasItem(objectLike(QUERY_DEBUG_FIELD, Query.class, queryFields)));
+
+        final List<FieldDebugInfo> paramSets = new ArrayList<>();
+
+        final List<FieldDebugInfo> paramSet1 = new ArrayList<>();
```

Comment:
I think it will be clearer if this list is build in another method, like in other tests above.